app/controllers/posts_controller.rb

```diff
@@ -6,8 +6,38 @@ def index
     @posts = @user.posts
   end
 
+  def new
+    @post = Post.new
+    @current_user = ApplicationController.current_user
+  end
+
+  def create
+    current_user = ApplicationController.current_user
+    @new_post =
+      Post.new(
+        author: current_user,
+        title: post_params['title'],
+        text: post_params['text'],
+        likes_counter: 0,
+        comments_counter: 0
+      )
```

Comment:
- [Optional] You did a great job defining the `post_params` method for security purposes :+1: You may refactor your code to something of this sort: `@new_post = @current_user.posts.new(post_params)`